OpenRA.Mods.Common/Activities/LayMines.cs

```diff
@@ -136,27 +147,42 @@ static bool CanLayMine(Actor self, CPos p)
 			return self.World.ActorMap.GetActorsAt(p).All(a => a == self);
 		}
 
-		void LayMine(Actor self)
+		bool StartLayingMine(Actor self)
 		{
 			if (ammoPools != null)
 			{
 				var pool = ammoPools.FirstOrDefault(x => x.Info.Name == minelayer.Info.AmmoPoolName);
 				if (pool == null)
-					return;
+					return false;
 
 				pool.TakeAmmo(self, minelayer.Info.AmmoUsage);
 			}
 
 			foreach (var t in self.TraitsImplementing<INotifyMineLaying>())
 				t.MineLaying(self, self.Location);
 
+			if (minelayer.Info.LayMineTime == 0)
+			{
+				FinishLayingMine(self);
+				return true;
+			}
+
+			return false;
+		}
+
+		void FinishLayingMine(Actor self)
+		{
+			if (minelayer.Info.MineLaidDelay > 0)
+				QueueChild(new Wait(minelayer.Info.MineLaidDelay)); // A little wait after placing each mine, for show
+			minefield.Remove(self.Location);
```

Comment:
```suggestion
			if (minelayer.Info.MineLaidDelay > 0)
				QueueChild(new Wait(minelayer.Info.MineLaidDelay));

			minefield.Remove(self.Location);
```

I don't think the comment is useful here.